src/CentreonRemote/Domain/Exporter/ConfigurationExporter.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/*
+ * Copyright 2005 - 2019 Centreon (https://www.centreon.com/)
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+namespace CentreonRemote\Domain\Exporter;
+
+use CentreonRemote\Infrastructure\Export\ExportManifest;
+use CentreonRemote\Infrastructure\Service\ExporterServiceAbstract;
+use Centreon\Domain\Repository;
+
+require_once __DIR__ . '/../../../../bootstrap.php';
+require_once 'config-generate-remote/generate.class.php';
+
+use ConfigGenerateRemote\Generate;
+use ConfigGenerateRemote\Manifest;
+
+class ConfigurationExporter extends ExporterServiceAbstract
+{
+
+    const NAME = 'configuration';
+    const MEDIA_PATH = _CENTREON_PATH_ . 'www/img/media';
+
+    /**
+     * Export data
+     */
+    public function export(int $remoteId): array
+    {
+        // create path
+        $this->createPath();
+
+        // call to ConfigGenerateRemote\Generate class
+        $dependencyInjector = loadDependencyInjector();
+        $config_generate = new Generate($dependencyInjector);
+        $config_generate->configRemoteServerFromId($remoteId, 'user');
+
+        return Manifest::getInstance($dependencyInjector)->getManifest();
+    }
+
+    /**
+     * Import data
+     *
+     * @param \CentreonRemote\Infrastructure\Export\ExportManifest $manifest
+     */
+    public function import(ExportManifest $manifest): void
+    {
+        // skip if no data
+        if (!is_dir($this->getPath())) {
+            return;
+        }
+
+        $db = $this->db->getAdapter('configuration_db');
+
+        // start transaction
+        $db->beginTransaction();
+
+        try {
+            // allow insert records without foreign key checks
+            $db->query('SET FOREIGN_KEY_CHECKS=0;');
+
+            $import = $manifest->get("import");
+            foreach ($import[data] as $data) {
+                // truncate table
+                $db->query("TRUNCATE TABLE `" . $data[table] . "`;");
+
+                // insert data
+                $exportPathFile = $this->getFile($data[filename]);
+                echo date("Y-m-d H:i:s") . " - INFO - Loading '" . $exportPathFile . "'.\n";
```

Comment:
Do not printing message directly from the method.
For example you can inject a logger class instance  and use it to log message.